overlord/servicestate/quota_control.go

```diff
@@ -70,51 +70,49 @@ func ensureSnapServicesForGroup(st *state.State, grp *quota.Group, allGrps map[s
 	return wrappers.EnsureSnapServices(snapSvcMap, ensureOpts, nil, progress.Null)
 }
 
-func validateSnapForAddingToGroup(st *state.State, name string, allGrps map[string]*quota.Group) error {
-	// validate that the snap exists
-	_, err := snapstate.CurrentInfo(st, name)
-	if err != nil {
-		return err
-	}
+func validateSnapForAddingToGroup(st *state.State, snaps []string, group string, allGrps map[string]*quota.Group) error {
+	for _, name := range snaps {
+		// validate that the snap exists
+		_, err := snapstate.CurrentInfo(st, name)
+		if err != nil {
+			return fmt.Errorf("cannot use snap %q in group %q: %v", name, group, err)
+		}
 
-	// check that the snap is not already in a group
-	for _, grp := range allGrps {
-		if strutil.ListContains(grp.Snaps, name) {
-			return fmt.Errorf("snap already in quota group %q", grp.Name)
+		// check that the snap is not already in a group
+		for _, grp := range allGrps {
+			if strutil.ListContains(grp.Snaps, name) {
```

Comment:
Per the comment in my earlier PR, I think we may reconsider this in the future followup and optimize to make a lookup map first. But it's ok for now.